pylint/checkers/classes/class_checker.py

```diff
@@ -880,6 +887,39 @@ def visit_classdef(self, node: nodes.ClassDef) -> None:
         self._check_proper_bases(node)
         self._check_typing_final(node)
         self._check_consistent_mro(node)
+        self._check_declare_non_slot(node)
+
+    def _check_declare_non_slot(self, node: nodes.ClassDef) -> None:
+        if not self._has_valid_slots(node):
+            return
+
+        slot_names = self._get_classdef_slots_names(node)
+
+        for base in node.bases:
+            ancestor = safe_infer(base)
+            if not isinstance(ancestor, nodes.ClassDef):
+                continue
+            if not self._has_valid_slots(ancestor):
+                return
+            else:
+                slot_names.extend(self._get_classdef_slots_names(ancestor))
```

Comment:
@Pierre-Sassoulas Look at our fancy output :smile:

_Unrelated to this PR but this is the first time I see these annotations in the wild :)_

Reply:
I'm not sure I understood 😄 I don't see annotations here ?